### PR TITLE
feat: update subnet table layout

### PR DESCRIPTION
**Pull request by Jay-Topher**
## Done
- Update fabric and space tables to support grouping

## QA steps

- [ ] visit subnets page
- [ ] Ensure that both fabric and space tables have row grouping

## Fixes

Fixes: [https://warthogs.atlassian.net/browse/MAASENG-2435](https://warthogs.atlassian.net/browse/MAASENG-2435)

### Fabric
image

### Space
image

### CODE REVIEW

File: src/app/base/components/GroupColumn/GroupColumn.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import type { Meta } from "@storybook/react";
+
+import GroupColumn from "./GroupColumn";
+
+const meta: Meta<typeof GroupColumn> = {
+  title: "Components/GroupColumn",
+  component: GroupColumn,
```
Comment (petermakowski) [inline src/app/base/components/GroupColumn/GroupColumn.stories.tsx:7]: This isn't really a column, but a group row. Please rename.

**Review by petermakowski — CHANGES_REQUESTED** [review]
A few things to improve below.

The items count in each group row should be the total number of rows in this group across all pages, not just the current page.

Rows nested in a group should have increased indentation, like in the machine list.

I'm pretty sure that each group name in the row group should be a link (a link to fabric or space depending on the grouping). Could you talk to @amylily1011 or @dgtlntv and verify this, plus any potential design considerations.

File: cypress/e2e/with-users/subnets/subnets.spec.ts
Comment (ndv99) [inline cypress/e2e/with-users/subnets/subnets.spec.ts:1]: Thanks for updating the Cypress tests for this!

File: src/app/subnets/views/SubnetsList/_index.scss [hunk 2]
```diff
@@ -61,4 +61,10 @@
       width: 1px;
     }
   }
+
+  .fabric-table, .space-table {
+    .subnet-row td:first-child {
+      padding-left: 1.8125rem;
```
Comment (ndv99) [inline src/app/subnets/views/SubnetsList/_index.scss:67]: Where does this value come from? I'd normally suggest using one of Vanilla's built-in spacing variables.

Comment (Jay-Topher) [reply src/app/subnets/views/SubnetsList/_index.scss]: I tried making the spacing, consistent with the machine list table, but this has been updated

**Review by ndv99 — APPROVED** [review]
This looks good IMO, nice one!

**Comment by ndv99** [comment]
> A few things to improve below.
> 
> The items count in each group row should be the total number of rows in this group across all pages, not just the current page.
> 
> Rows nested in a group should have increased indentation, like in the machine list.
> 
> I'm pretty sure that each group name in the row group should be a link (a link to fabric or space depending on the grouping). Could you talk to @amylily1011 or @dgtlntv and verify this, plus any potential design considerations.

Before merging - @Jay-Topher did you follow up on this?

**Comment by Jay-Topher** [comment]
> > A few things to improve below.
> > The items count in each group row should be the total number of rows in this group across all pages, not just the current page.
> > Rows nested in a group should have increased indentation, like in the machine list.
> > I'm pretty sure that each group name in the row group should be a link (a link to fabric or space depending on the grouping). Could you talk to @amylily1011 or @dgtlntv and verify this, plus any potential design considerations.
> 
> Before merging - @Jay-Topher did you follow up on this?

Yes, I did, you could verify also but I'm sure I covered everything there

File: src/app/subnets/views/SubnetsList/SubnetsTable/SubnetsTable.test.tsx [hunk 3]
```diff
@@ -435,3 +405,39 @@ it("remains on the same page once the data is updated and page is still availabl
...
+      name: "2",
+    })
+  );
+
+  await waitFor(() =>
+    expect(
+      screen
+        .getByRole("navigation", { name: "pagination" })
+        // eslint-disable-next-line testing-library/no-node-access
+        .querySelector(".is-active")
```
Comment (petermakowski) [inline src/app/subnets/views/SubnetsList/SubnetsTable/SubnetsTable.test.tsx:435]: you should be able to use `getByRole("button", { current: true })`

File: src/app/subnets/views/SubnetsList/SubnetsTable/components.tsx [hunk 4]
```diff
@@ -59,3 +70,143 @@ export const CellContents = ({
...
+            ),
+          },
+        ],
+      });
+    }
+
+    rows = rows.concat(generateSubnetRows(networks));
+  });
+  return rows;
+};
```
Comment (petermakowski) [inline src/app/subnets/views/SubnetsList/SubnetsTable/components.tsx:212]: There'a some repetition here that could be avoided which would also make this easier to understand. One way to do that below.


```suggestion
export const generateSubnetGroupRows = ({
  itemName,
  groups,
  columnLength,
  groupMap,
}: {
  itemName: string;
  groups: FabricTableRow[] | SpaceTableRow[];
  columnLength: number;
  groupMap: Record<string | number, { count: number }>;
}) => {
  const generateGroupRow = (name: string) => {
    return {
      "aria-label": `${name} group`,
      className: "",
      columns: [
        {
          colSpan: columnLength,
          content: (
            <GroupRow
              count={groupMap[name].count}
              groupName={name}
              itemName={itemName}
            />
          ),
        },
      ],
    };
  };

  return groups.flatMap((group) => {
    const { networks } = group;
    const name = "fabricName" in group ? group.fabricName : group.spaceName;
    return [generateGroupRow(name as string), ...generateSubnetRows(networks)];
  });
};
```

File: src/app/subnets/views/SubnetsList/SubnetsTable/utils.tsx [hunk 5]
```diff
@@ -40,50 +40,98 @@ const getColumn = (label: string | null, href?: string | null) => ({
...
+          result[item.sortData.spaceName].count += 1;
+        } else {
+          result[item.sortData?.spaceName] = { count: 1 };
+        }
+      }
+    });
+  }
+
+  return result;
+};
```
Comment (petermakowski) [inline src/app/subnets/views/SubnetsList/SubnetsTable/utils.tsx:133]: This could be made shorter and clearer similarly to `generateSubnetGroupRows`

**Pull request merged**